pwndbg/dbg/gdb.py

```diff
@@ -112,17 +87,79 @@ def __init__(self, inner: gdb.Frame):
 
     @override
     def evaluate_expression(self, expression: str) -> pwndbg.dbg_mod.Value:
+        from pwndbg.gdblib.scheduler import lock_scheduler
+
+        with lock_scheduler():
```

Comment:
The previous code doesn't seem to have this, why are we adding it here?

Reply:
Well, mostly so we can also cover the cases that previously needed `parse_and_eval_with_scheduler_locking`. I think there's probably a place for us to maybe add a `allow_resume_threads` keyword argument, but LLDB will most likely just ignore that, seeing as things like the Clang-based expression evaluator just uses a [thread plan on the thread that owns the frame](https://github.com/llvm/llvm-project/blob/1601879f5d690595889a5537c0049b62df4657c7/lldb/source/Expression/ExpressionParser.cpp#L62). Maybe that argument could be an optional optimization? As in: allow other threads to resume if it's faster, but with no guarantees that the debugger will actually honor it.

Reply:
My point is, scheduler locking can cause deadlocks sometimes, so its better if the user control it and we only use it if its really needed.

Reply:
Okay, fair enough. I wasn't aware of the issues with deadlocking. I'll add a PR to have this be conditionally enabled. We'll have to expose scheduler locking in the Debugger-agnostic API, but it's worth it, given the possible consequences of not doing it.